secant/Screens/Onboarding/OnboardingScreenViewModel.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  OnboardingScreenViewModel.swift
+//  secant-testnet
+//
+//  Created by Francisco Gindre on 9/17/21.
+//
+
+import Foundation
+import Combine
+
+protocol OnboardingStepProviding {
+    var totalSteps: Int { get }
+    var currentStep: OnboardingStep { get }
+    var hasNext: Bool { get }
+    var hasPrevious: Bool { get }
+    func next()
+    func previous()
+}
+
+struct OnboardingStep {
+    var title: String?
+    var imageName: String
+    var blurb: String
+    var stepNumber: Int
+}
+
+class OnboardingScreenViewModel: BaseViewModel<OnboardingStepProviding>, ObservableObject {
+    enum RightBarButton {
+        case skip
+        case close
+        case none
+    }
+    @Published var currentStep: OnboardingStep
+
+    var totalSteps: Int {
+        services.totalSteps
+    }
+
+    var showPreviousButton: Bool {
+        services.hasPrevious
+    }
+
+    var showNextButton: Bool {
+        services.hasNext
+    }
+
+    var showRightBarButton: RightBarButton {
+        services.hasNext ? .skip : .close
+    }
+
+    var showStepper: Bool {
+        services.totalSteps > 1
+    }
+
+    override init(services: OnboardingStepProviding) {
+        self.currentStep = services.currentStep
+        super.init(services: services)
+    }
+
+    func next() {
+        services.next()
+        self.currentStep = services.currentStep
+    }
+
+    func previous() {
+        services.previous()
+        self.currentStep = services.currentStep
+    }
+
+    func skip() {}
+}
+
+extension OnboardingStep: Equatable {
```

Comment:
This looks like it should be auto synthesized by the compiler. Did you try just adding the conformance directly to the type instead of manually creating this?

Reply:
I did when It was on the testing target :( it did not auto synthesized it :'(

Reply:
Now it did  🤔